zap-widget/src/components/SearchBox.tsx

```diff
@@ -0,0 +1,188 @@
+  warn?: boolean
+  hint?: string
+  lnAddress?: string
+}
+
+function SearchBox() {
+  const [packageQueryName, setPackageQueryName] = useState('')
+  const [result, setResult] = useState<ResultType | null>(null)
+  const [isLoading, setIsLoading] = useState(false)
+  const [amountInSats, setAmountInSats] = useState(0)
+
+  const { invoice, setInvoice } = useClient()
+
+  useEffect(() => {
+    const decodeInv = () => {
+      if (invoice.length > 1) {
+        const amountInSat = decode.decode(invoice) as { sections: { value?: string | number }[] }
+        setAmountInSats(+amountInSat.sections[2]?.value / 1000)
+      }
+    }
+
+    decodeInv()
+  }, [invoice])
+
+  const fetchPackage = async () => {
+    try {
+      //clear the result and input alone - when the cancel icon is clicked
+      if (result) {
+        setResult(null)
+        setPackageQueryName('')
+        return
+      }
+      setInvoice('')
+      setAmountInSats(0)
+
+      //fetch the package
+      setIsLoading(true)
+      if (packageQueryName.length > 1) {
+        const response = await fetch(`https://registry.npmjs.org/${packageQueryName}/latest`)
+        const packageInfo = await response.json()
+
+        if (!packageInfo || packageInfo === 'Not Found') {
+          setResult({
+            warn: true,
+            packageName: packageQueryName,
+            hint: '⚠️ Can`t find this package. Typo or doesn`t exist? 🤔',
+          })
+          return
+        } else if (!packageInfo.funding || packageInfo.funding.type !== 'lightning') {
+          setResult({
+            warn: true,
+            packageName: packageQueryName,
+            hint: '⚠️ This package does not accept bitcoin tips yet 🥺',
+          })
+          return
+        } else {
+          const lnAddress = packageInfo.funding.url
+
+          setResult({
+            warn: false,
+            lnAddress: lnAddress.startsWith('lightning:') ? lnAddress.substring(10) : lnAddress,
+            packageName: packageInfo.name,
+            description: packageInfo.description,
+          })
+        }
+      }
+    } catch {
+      setResult({
+        warn: true,
+        hint: 'Something went wrong, please try again :(',
+      })
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen mx-8">
+      <div className="w-full flex items-center justify-center flex-col md:flex-row gap-2 md:gap-6">
+        <input
+          value={packageQueryName}
+          onChange={(e) => {
+            if (result) {
+              setResult(null)
+            }
+            setPackageQueryName(e.target.value)
+          }}
+          placeholder="Search package name..."
+          className="w-full md:max-w-xl h-12 md:h-[72px] rounded-full pl-5 py-2 bg-zap-gradient border border-white/25 text-2xl font-grotesk placeholder-opacity-70 outline-none"
+        />
+        <button
+          onClick={fetchPackage}
```

Comment:
You can split this into two buttons and render them like `result ? cancel <button> : go <button>`